### DISCUSSION TITLE
Additional font glyphs

### DISCUSSION THREAD

**Discussion by dstroud** [monome/norns#1774] [General]
Any interest in including some additional glyphs in the default font? I've had a go at some superscripts that might be useful for chords, as well as some media playback and misc. glyphs. 

Issues to think through:
- Not all alphanumeric characters can be well represented as 4-pixel height superscript. I think the ones most useful for chord naming (i.e. 7th, flat 5, and b superscript as a sub for ♭) are okay but some characters would be rather abstract (or could be omitted).
- Some nice-to-have characters like superscript ø, ♭, ♯, and ♮ don't have standard code points so would likely need to use the _Private Use Area in the Basic Multilingual Plane_ (starting at code point U+E000).
- There is some overlap with the `UI` lib and the glyphs differ in an effort to match weights and include fast-forward/backward.

One nice aspect of this approach is that these can be used in the initial script comments/description. See attached screenshot  which is rendered from the following:

```lua
-- Std.supers: ¹²³⁴⁵⁶⁷⁸⁹⁰⁺⁻⁽⁾ᴹᵐᵃʲᵇ
-- Private-use supers: 
-- Media: ⏵ ⏸ ⏹ ⏺ ⏴ ⏪ ⏩
-- Romans: ⅰ ⅱ ⅲ ⅳ ⅴ ⅵ ⅶ
--          Ⅰ Ⅱ Ⅲ Ⅳ Ⅴ Ⅵ Ⅶ
-- Extras: ∆✓
-- Ex: Cmᴹ⁷  Cm7ᵇ⁵  iᵐᵃʲ⁷  Ⅳᴹ⁷
```

![fonts_240422](https://github.com/monome/norns/assets/435570/a354af10-7d76-4944-95af-ad0a1abd534c)

**Comment by tehn** [comment]
this seems like a good idea--- new functionality that doesn't break anything backwards, so is there a downside?

**Reply by dstroud** [reply]
Thanks for weighing in.  I don't see much downside-- some small increase in memory usage perhaps, although the file size of the WIP seems to have actually gotten smaller somehow!

Some of the superscript characters can be a bit cryptic/ugly (see updated image above that includes _all_ numerical superscripts rather than just the acceptable 5 and 7 that seem most useful for chord names). I'd be fine dropping those if they seem an inelegant proposal.

Another thing to consider: how to best approach such additions while being respectful of the original font artist's creation. Feels like a Ship of Theseus thought experiment; is it still the same font if heavily extended?

One approach could be creating a whole new font, pixel-by-pixel, based on the existing one. That could include attribution to the original artist in the metadata while making it clear that this is a separate, derivative work.

The artist's [translated] FAQ (from 20+ years ago!) suggests no issues with that approach:

_Q. Can I create *** using font 04?
A. Other than reselling the font itself (font data), any use, whether private or commercial, is free. I'm really happy that you can use it. thank you. (This does not mean that you are abandoning the copyright, but you do not need to include the copyright notice.)_

I see creating a new font specifically for Norns as accomplishing a few nice things:
1. Properly attributing the original artist (no copyright¹, etc... seems to exist in the current .ttf metadata)
2. Flexibility to modify the font as needed in the future without impinging on the original art
3. New .ttf filename which may be useful in light of https://github.com/monome/norns/issues/1773

I am, of course, happy to volunteer my time if there is interest in this.

¹ Some [quick research](https://en.wikipedia.org/wiki/Intellectual_property_protection_of_typefaces) suggests copyright does not actually cover fonts in US/JP so I think this is largely a matter of courtesy.

**Comment by tehn** [comment]
apologies for the delay--- if you'd like to work on this i'd be happy to see the contribution.

**Comment by Dewb** [comment]
Just wanted to add that the font has already been extended (a few times) and we talked about some of those issues (not definitively) in previous changes: https://github.com/monome/norns/pull/919

**Comment by dstroud** [comment]
I have a WIP .ttf and some resources available at https://github.com/dstroud/norns-font/

@Dewb, thanks so much for your work extending the font to this point. I made some notes on my proposal to tweak 3 existing glyphs in the .pdf but  wouldn't feel comfortable moving forward with those changes without your approval as our resident pixel-font rock star.

<img width="389" alt="Screenshot 2024-05-05 at 7 24 36 PM" src="https://github.com/monome/norns/assets/435570/96dc61e9-dcc6-4faf-900a-089dc9e08474">

**Reply by tyleretters** [reply]
i love this
